boxmaker_demo.scad

```diff
@@ -0,0 +1,112 @@
+use <lib/boxmaker.scad>;
+
+// Box inner dimensions
+box_inner = [100, 70, 50];
+// Material thickness (mm)
+thickness = 3;
+
+// Tab width (X, Y, Z, TopX, TopY) (mm)
+// The TopX and TopY here create a nice easy-to-open lid.
+tabs_with_lid = [7, 7, 7, box_inner[0] / 3, box_inner[1] / 3];
+
+box_with_lid(box_inner, thickness, tabs_with_lid);
+
+// When there's no top, leave off the last two dimensions
+// Tab width (X, Y, Z) (mm)
+tabs_no_top = [7, 7, 7];
+
+translate([box_inner[0] + 20, 0, 0])
+  box_without_top(box_inner, thickness, tabs_no_top);
+
+module box_with_lid(box_inner, thickness, tabs) {
+  // Due to the limitations of OpenSCAD, the best way to allow for the same
```

Comment:
Does it make sense to use consistent naming conventions for things like tabs/fingers? I think It makes sense to choose one. This style of joint in wood working is commonly called a comb joint or a finger joint.

Reply:
Yeah, good call. Let's stick with "fingers"